libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -357,6 +359,19 @@ public void run() {
         return view;
     }
 
+    private void initializeSavingProgressDialog() {
+        if (mEditorFragmentListener != null) {
+            mEditorFragmentListener.getSavingInProgressDialogVisibility()
+                                   .observe(getViewLifecycleOwner(), visibility -> {
+                                       if (DialogVisibility.Showing == visibility) {
+                                           showSavingProgressDialogIfNeeded();
+                                       } else {
+                                           hideSavingProgressDialog();
```

Comment:
Having hiding the dialog here in addition to hiding it in `onDestroy` seems to be a bit of double safety (i.e., we don't have to hide it in both places), but I thought it made sense to do it in both places since it is a blocking dialog.

Reply:
Yep, I agree ;). Thanks